sgs/api.py

```diff
@@ -22,3 +23,35 @@ def get_data(ts_code: int, begin: str, end: str) -> List:
     request_url = url.format(ts_code, begin, end)
     response = requests.get(request_url)
     return response.json()
+
+
+@retry(stop_max_attempt_number=MAX_ATTEMPT_NUMBER)
+@functools.lru_cache(maxsize=LRU_CACHE_SIZE)
+def get_data_olinda(resource: str, begin: str, end: Union[None, str]) -> List:
+    """
+    Requests data frames from the BCB Olinda API in json format.
+    """
+    # resource names
+    names = {
+        'monthly': 'ExpectativaMercadoMensais',
+        'quarterly': 'ExpectativaMercadoTrimestrais',
+        'yearly': 'ExpectativaMercadoAnuais',
+        'infl12m': 'ExpectativasMercadoInflacao12Meses',
+        'monthlytop5': 'ExpectativasMercadoTop5Mensais',
+        'yearlytop5': 'ExpectativasMercadoTop5Anuais',
+        'inst': 'ExpectativasMercadoInstituicoes'
+        }
+
+    # TODO: Olinda API expects ISO dates
+    begin_iso = datetime.datetime.strptime(begin, '%d/%m/%Y').strftime('%Y-%m-%d')
+    end_iso = datetime.datetime.strptime(end, '%d/%m/%Y').strftime('%Y-%m-%d')
```

Comment:
Sugiro alterar o seguinte trecho para referenciar a função já importada:

```suggestion
    begin_iso = to_datetime(begin, 'pt')
    end_iso = to_datetime(end, 'pt')
```